src/downloadSinglePost.ts

```diff
@@ -1,73 +1,84 @@
-import needle from "needle";
-import cliProgress from "cli-progress";
-import fs from "fs";
-import path from "path";
-import { VliveDownloader } from "./vliveDownloader";
+import needle from 'needle';
+import cliProgress from 'cli-progress';
+import fs from 'fs';
+import path from 'path';
+import { VliveDownloader } from './vliveDownloader';
 
-const downloadSinglePost = async (vliveUrl: string) => {
+const downloadSinglePost = async (vliveUrl: string, callback: any) => {
   try {
-    const isVliveUrl = vliveUrl.includes("vlive.tv/post");
-    let postId = "";
+    const isVliveUrl = vliveUrl.includes('vlive.tv/post');
+    let postId = '';
 
-    if (isVliveUrl) {
-      postId = vliveUrl.split("/").pop() as string;
-    } else {
-      throw new Error("Not vlive url");
-    }
+    // if (isVliveUrl) {
+    //   // postId = vliveUrl.split('/').pop() as string;
+    // } else {
+    //   throw new Error('Not vlive url');
+    // }
+
+    if (!isVliveUrl) throw new Error('Not vlive url');
+    postId = vliveUrl.split('/').pop() as string;
 
     const downloader = new VliveDownloader(postId);
     await downloader.getPostData();
     await downloader.getVideoKey();
     await downloader.getVideoMetaData();
     downloader.withCategory();
 
-    const validPath = path.normalize(downloader.videoData.data.title.replace(/[^a-zA-Z0-9 ]/g, ""));
-    if (!fs.existsSync(validPath)) {
-      fs.mkdirSync(validPath);
+    // const mainPath = path.normalize(downloader.videoData.data.title.replace(/[^a-zA-Z0-9 ]/g, ''));
+
+    const dateNamePath = new Date(downloader.videoData.data.createdAt).toISOString().substr(0, 10) + ` - ${Math.random().toString(23).substring(2, 6)}`;
```

Comment:
Better to include the file name in the folder name, instead of using a random string.
```suggestion
    const sanitizeString = (str: string) => path.normalize(str.replace(/[`~!@#$%^&*_|+\-=?;:'",.<>{}\/]/gi, ""));
    const dateNamePath =
      new Date(downloader.videoData.data.createdAt).toISOString().substring(0, 10) + " " + sanitizeString(downloader.videoData.data.title);
```

Also, `substr()` is deprecated, use [substring()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr) instead.